packages/rn-tester/js/examples/ActivityIndicator/ActivityIndicatorExample.js

```diff
@@ -10,46 +10,37 @@
 
 import type {Node} from 'React';
 import {ActivityIndicator, StyleSheet, View} from 'react-native';
-import React, {Component} from 'react';
+import React, {useState} from 'react';
+import {useEffect} from 'react';
 
-type State = {|animating: boolean|};
-type Props = $ReadOnly<{||}>;
-type Timer = TimeoutID;
+function ToggleAnimatingActivityIndicator() {
+  let _timer;
 
-class ToggleAnimatingActivityIndicator extends Component<Props, State> {
-  _timer: Timer;
+  const [animating, setAnimating] = useState(true);
 
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      animating: true,
-    };
-  }
-
-  componentDidMount() {
-    this.setToggleTimeout();
-  }
+  const setToggleTimeout = () => {
+    _timer = setTimeout(() => {
+      setAnimating(currentState => !currentState);
+      setToggleTimeout();
+    }, 2000);
+  };
 
-  componentWillUnmount() {
-    clearTimeout(this._timer);
-  }
+  useEffect(() => {
+    setToggleTimeout();
 
-  setToggleTimeout() {
-    this._timer = setTimeout(() => {
-      this.setState({animating: !this.state.animating});
-      this.setToggleTimeout();
-    }, 2000);
-  }
+    return () => {
+      clearTimeout(_timer);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
What warning does this rule give? I don't think we should disable it.

Reply:
It gives and error related the setToggleTimeout, I've just fixed it